crates/rattler_lock/src/pypi.rs

```diff
@@ -1,35 +1,79 @@
 use crate::PackageHashes;
+use pep440_rs::VersionSpecifiers;
+use pep508_rs::Requirement;
 use serde::{Deserialize, Serialize};
 use serde_with::{serde_as, skip_serializing_none};
+use std::cmp::Ordering;
 use std::collections::HashSet;
 use url::Url;
 
-/// A pinned PyPi package
+/// A pinned Pypi package
 #[serde_as]
 #[skip_serializing_none]
-#[derive(Serialize, Deserialize, Eq, PartialEq, Clone, Debug)]
-pub struct PypiLockedDependency {
-    /// A list of dependencies on other packages that the wheel listed.
-    #[serde(default, alias = "dependencies", skip_serializing_if = "Vec::is_empty")]
-    #[serde_as(deserialize_as = "crate::utils::serde::Pep440MapOrVec")]
-    pub requires_dist: Vec<String>,
+#[derive(Serialize, Deserialize, Eq, PartialEq, Clone, Debug, Hash)]
+pub struct PypiPackageData {
+    /// The name of the package.
+    pub name: String,
 
-    /// The python version that this package requires.
-    pub requires_python: Option<String>,
-
-    /// A list of extras that are selected
-    #[serde(default, skip_serializing_if = "HashSet::is_empty")]
-    pub extras: HashSet<String>,
+    /// The version of the package.
+    pub version: pep440_rs::Version,
 
     /// The URL that points to where the artifact can be downloaded from.
     pub url: Url,
 
     /// Hashes of the file pointed to by `url`.
+    #[serde(flatten)]
     pub hash: Option<PackageHashes>,
 
-    /// ???
-    pub source: Option<Url>,
+    /// A list of dependencies on other packages that the wheel listed.
```

Comment:
Does this also count for the sdists?

Reply:
Yes, metadata is gathered from sdists in the same way as wheels.

Reply:
I removed the `"wheel"` part of the documentation.